google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/DefaultCloudSdkService.java

```diff
@@ -38,6 +41,9 @@ public String getCloudSdkHomePath() {
 
   @Override
   public void setCloudSdkHomePath(String cloudSdkHomePath) {
-    propertiesComponent.setValue(CLOUD_SDK_PROPERTY_KEY, cloudSdkHomePath);
+    if (!StringUtil.isEmpty(cloudSdkHomePath)
+        && CloudSdkUtil.containsCloudSdkExecutable(cloudSdkHomePath)) {
```

Comment:
:+1: for moving this logic into the `DefaultCloudSdkService` - definitely feels like a better place for it. I know it's pretty simple, but I might suggest adding a small test for this logic. I wouldn't consider it crucial though.